src/System.Private.CoreLib/shared/System/ArraySegment.cs

```diff
@@ -175,7 +175,7 @@ public T[] ToArray()
             get
             {
                 ThrowInvalidOperationIfDefault();
-                if (index < 0 || index >= _count)
+                if ((uint)index >= (uint)_count)
```

Comment:
The JIT would not be able to optimize cases like this. It does not know that _count is not negative.

Reply:
@jkotas what about `array.Length`? Does JIT know it can't be negative? (e.g. https://github.com/mono/mono/issues/16948)

Reply:
Also, clang still is able to optimize it: https://godbolt.org/z/nmC3c6

Reply:
Clang's Isn't as optimized?
```asm
    cmp     edi, esi
    setg    al
    shr     edi, 31
    or      rdi, rax
    jne     .LBB0_1
```
vs [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4axAMxNSDIQwDeNGgzNMUDACqqMpABS8eDXBmxQMASS4ATGAjQGJwwGaS4AcwwACwBKU3NjanNkoIAzBnt7DmCY13cvX38GAD4M7J4YsMjY+JSjWrrzADEICHJ7GI0kxoBfBuSWiAdOhr7uswbiS0H2mMTGhmioCAB3Bi4YNYBRBBUxPgguDrVzUYaGgG0AWRhoiB9PcUl7G7uHp4B5fd5D3BYAOQg3kkTic4RiAF1JtNWsNDAwxj0gA=)
```asm
    cmp edx, r8d
    ja L001a
```

Reply:
clang doesn't know our integer is actually a positive number :-)